File: fifteen-exercises/h-index.rs
impl Solution {
    pub fn h_index(citations: Vec<i32>) -> i32 {
        let mut citations = citations;
        citations.sort();
        citations.reverse();
        let mut h = 0i32;
        let mut cnt = 0;
        for citation in citations {
            cnt += 1;
            if citation >= cnt {
                h = cnt
            }
        }
        h
    }
}

struct Solution;

fn main() {
    assert_eq!(Solution::h_index(vec![3,0,6,1,5]), 3);
    assert_eq!(Solution::h_index(vec![0]), 0);
    assert_eq!(Solution::h_index(vec![1]), 1);
    println!("ok")
}

File: fifteen-exercises/h-index.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int hIndex(vector<int> &citations) {
        sort(citations.begin(), citations.end());
        int h = 0;
        int cnt = 0;
        for (auto i = citations.rbegin(); i != citations.rend(); ++i) {
            ++cnt;
            if (*i >= cnt) {
                h = cnt;
            }
        }
        return h;
    }
};

int main() {
    cout << "ok" << endl;
    return 0;
}
